### PR TITLE
Adds profile type

### PR DESCRIPTION
**Pull request by krishguptadev**
Done 1st and 2nd thing listed in https://github.com/EddieHubCommunity/LinkFree/issues/696#issuecomment-979972785

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/749"></a>

### CODE REVIEW

**Comment by Vyvy-vi** [comment]
I think it would make sense to keep `type` as an optional parameter

**Comment by krishguptadev** [comment]
> I think it would make sense to keep `type` as an optional parameter...
> 
> Due to which, I think we can omit that key from the personal profiles.(and have it default to "personal")

Eddie on discord decided to keep it a required field? (I asked them twice)

**Comment by ssensalo** [comment]
Wow this is a lot of work done, thanks @krishguptadev

**Review by ssensalo — APPROVED** [review]

File: public/data/lailahgrant.json [hunk 1]
```diff
@@ -18,28 +19,5 @@
       "url": "https://www.linkedin.com/in/lailah-grant-561692168/",
       "icon": "linkedin"
     }
-  ],
-  "milestone": [
```
[the remark this answers is not in the document]
Comment (krishguptadev) [reply public/data/lailahgrant.json:22]: This got removed because this was the wrong key, it should be `milestones`. Will fix that issue here.

**Review by Panquesito7 — APPROVED** [review]

**Comment by Vyvy-vi** [comment]
> Eddie on discord decided to keep it a required field? (I asked them twice)

As far as I remember the discussion, we talked about keeping it an optional param, so that it doesn't cause unnecessary inconvenience

Idk, it would be advantageous to default "type" to "personal"/"profile", so that there isn't extra boilerplate needed to add a file...
Moreover, as mentioned in https://github.com/EddieHubCommunity/LinkFree/issues/696#issuecomment-979972785 -
> I think if we have an optional type field in the json for community | personal

**Comment by krishguptadev** [comment]
shall I revert?

**Review by emmalearnscode — APPROVED** [review]

**Review by shreyaansjain06 — APPROVED** [review]

**Review by schmelto — APPROVED** [review]

**Pull request merged**